src/external/wb_presets.c

```diff
@@ -5227,6 +5227,13 @@ const wb_data wb_preset[] =
   { "Panasonic", "DC-G9", Incandescent, 0,      { 1.632813, 1, 2.839844, 0 } },
   { "Panasonic", "DC-G9", Flash, 0,             { 2.566406, 1, 1.734375, 0 } },
 
+  /* Panasonic DC-G90 Firmware Version 1.0 */
+  { "Panasonic", "DC-G90", Daylight, 0,                { 2.226563, 1, 1.902344, 0 } },
+  { "Panasonic", "DC-G90", Cloudy, 0,                  { 2.367188, 1, 1.777344, 0 } },
+  { "Panasonic", "DC-G90", Shade, 0,                   { 2.511719, 1, 1.679688, 0 } },
+  { "Panasonic"; "DC-G90", Incandescent, 0,            { 1.601563, 1, 2.859375, 0 } },
```

Comment:
You have a ; here instead of , !

Have you tested that?

Reply:
I had to make some manual changes because the [wb_extract script](https://github.com/sergiomb2/ufraw/blob/master/wb_extract.pl) I used didn't work properly. I made sure to test it properly this time. I realized there was a script in the darktable repo and used that one now but it appears that it is not able to get the name of the preset.

> mikael:~/wd/darktable/tools% ./extract_wb P1001028.RW2
> WARNING: no defined sort order for '2'
> WARNING: no defined sort order for '3'
> WARNING: no defined sort order for '1'
> WARNING: no defined sort order for '2'
> WARNING: no defined sort order for '4'
> WARNING: no defined sort order for '5'
> WARNING: no defined sort order for '6'
> WARNING: no defined sort order for '7'
> WARNING: no defined sort order for '4'
> WARNING: no defined sort order for '6'
> WARNING: no defined sort order for '5'
> WARNING: no defined sort order for '6'
> WARNING: no defined sort order for '1'
> WARNING: no defined sort order for '4'
> WARNING: no defined sort order for '2'
> WARNING: no defined sort order for '4'
> WARNING: no defined sort order for '3'
> WARNING: no defined sort order for '4'
>  { "Panasonic", "DC-G95", 1, 0, { 2.2265625, 1, 1.90234375, 0 } },
>  { "Panasonic", "DC-G95", 2, 0, { 2.3671875, 1, 1.77734375, 0 } },
>  { "Panasonic", "DC-G95", 3, 0, { 2.51171875, 1, 1.6796875, 0 } },
>  { "Panasonic", "DC-G95", 4, 0, { 1.6015625, 1, 2.859375, 0 } },
>  { "Panasonic", "DC-G95", 5, 0, { 2.3046875, 1, 1.8671875, 0 } },
>  { "Panasonic", "DC-G95", 6, 0, { 2.09765625, 1, 1.94921875, 0 } },
>  { "Panasonic", "DC-G95", 7, 0, { 1.6015625, 1, 2.859375, 0 } },


> mikael:~% perl wb_extract.pl P1001028.RW2
> Use of uninitialized value $model in sprintf at wb_extract_new.pl line 132.
>   { "DC-G90", "", "Daylight", 0,                { 2.226563, 1, 1.902344, 0 } },

The presets do not show up with DC-G90 as camera so I guess darktable reads a different value. After changing it to DC-G95 the presets are showing up.

![Anteckning 2020-01-20 102425](https://user-images.githubusercontent.com/5606160/72715650-28cba780-3b71-11ea-94fa-096641e4fc48.png)

Reply:
If the camera is really a G90 renaming to G95 is wrong. What is the name of the camera displayed in the image information module?

Reply:
![2020-01-20 140545](https://user-images.githubusercontent.com/5606160/72728603-08f6ac80-3b8e-11ea-9555-90cca4dfbe58.png)

The name in the Image Information is DC-G90 and the camera is a DC-G90

What could cause the presets to only appear when the model is changed to DC-G95 in wb_presets.c?